### DISCUSSION TITLE
Alternative to e3nn.networks.GatedConvParityNetwork

### DISCUSSION THREAD

**Discussion by CorneCorne** [e3nn/e3nn#303] [Q&A]
Hello.

If I'm correct, the e3nn.networks module was erased from e3nn in January 2021.

[Question for handling different size of input via masking #104](https://githubmemory.com/repo/e3nn/e3nn/issues/104)

In the code at the above URL

`from e3nn.networks import GatedConvParityNetwork`

Is there an alternative module or class? It causes an import error in the current e3nn.

I would also like to handle data where each data has a different number of coordinates like the questioner in the above URL.

Thanks.

**Comment by CorneCorne** [comment]
Perhaps I've done. I think it is one of the smallest codes that constitutes a dataset.

```
# (6,3)
# coordinate (x, y, z)
position_list = [[-4.5037 -1.6629 -3.3617],
 [-4.3601 -1.6275 -3.3653],
 [-4.3321 -1.5348 -3.2489],
 [ 3.7348 -0.9221 -0.4269],
 [ 3.6691 -1.073  -0.6121],
 [ 3.3018 -1.2151 -0.2409]]

# (6,5)
feature_list = [[0. 0. 1. 0. 0.],
 [0. 1. 0. 0. 0.],
 [0. 1. 0. 0. 0.],
 [0. 1. 0. 0. 0.],
 [0. 1. 0. 0. 0.],
 [0. 0. 0. 1. 0.]]
class MyDataset(torch_geometric.data.Dataset):
    def __init__(self):
        self.data_list = []
        for i in range(len(position_list)):
            position = position_list[i]
            feature = feature_list[i]
            data = torch_geometric.data.Data(pos=torch.tensor(position, dtype=dtype))
            data.input_element = torch.tensor(feature , dtype=dtype)
            self.data_list.append(data)
   
    def __len__(self):
        return len(self.data_list)
    def __getitem__(self, idx):
        return self.data_list[idx]

```


BTW, I don't fully understand the background of e3nn.
In CNN, we generally store objects in voxels, but I don't think e3nn uses voxels to treat them like graphs. At least in my code, I don't consider the length of the voxel. However, if the data is very large on voxels, my model may not learn well.

**Comment by mariogeiger** [comment]
We use the mini-batching way of `pytorch_gemetric` to do that. see https://pytorch-geometric.readthedocs.io/en/latest/notes/batching.html
Here you can see an example where we use it https://docs.e3nn.org/en/stable/guide/periodic_boundary_conditions.html
